src/client/pages/forgot-password.js

```diff
@@ -0,0 +1,119 @@
+import React, { PureComponent } from 'react';
+
+import Error from '~/components/error';
+import Button from '~/components/button';
+import Layout from '~/layouts/main';
+import { setFormValue } from '~/helpers';
+import { forgotPassword } from '~/resources/account/account.api';
+
+export default class Signin extends PureComponent {
+  constructor(...args) {
+    super(...args);
+    this.setEmail = setFormValue('email').bind(this);
+  }
+
+  componentWillMount() {
+    this.state = {
+      email: '',
+      emailSent: false,
+
+      isLoading: false,
+
+      error: null,
+    };
+  }
+
+  submitSignin = async (event) => {
+    event.preventDefault();
+    try {
+      this.setState({ isLoading: true, error: null });
+      await forgotPassword({ email: this.state.email });
+      this.setState({ emailSent: true });
+    } catch (error) {
+      this.setState({ error });
+    } finally {
+      this.setState({ isLoading: false });
```

Comment:
I think that here we can use less calls of the `setState` function. Each call to the setState function calls render of the component, so we must avoid unnecessary calls.
```javascript
this.setState({ isLoading: true, error: null });
const newState = {};
try {
  await forgotPassword({ email: this.state.email });
  newState.emailSent = true;
} catch (error) {
  newState.error = error;
} finally {
  this.setState({
    ...newState,
    isLoading: false,
  })
}
```